### ISSUE TITLE
[Feature Request] Add Deadline Option in FSRS to Optimize Reviews for Specific Dates (Exams)

### ISSUE THREAD

**Issue by cjdduarte** [open-spaced-repetition/fsrs4anki-helper#456]
**Is your feature request related to a problem? Please describe.**

I was reading a discussion on Reddit that brought up some interesting points about how spaced repetition could be optimized for exams with set dates. One of the main topics was the possibility of adjusting FSRS to maximize retention on the exam day, even if that means allowing for lower retention leading up to it. The current system focuses on maintaining retention over time, but some users need a more flexible approach that targets specific deadlines, like exams.

**Describe the solution you'd like**

It would be helpful to have an option in FSRS that allows users to set a **deadline** or target date, so the algorithm can adjust review schedules to ensure maximum retention on that day. This could involve initially using longer intervals and then concentrating reviews as the deadline approaches.

**Describe alternatives you've considered**

Currently, users could manually adjust review intervals or add custom reviews before the exam. However, this is less efficient and goes against the automatic optimization FSRS provides. A feature to set a deadline would make this process much smoother and more efficient.

**Additional context**

Here’s an image that illustrates the idea of focusing reviews toward a specific date:  

![image](https://github.com/user-attachments/assets/c8cf0fba-2ffb-4a1a-ab7b-441435c4b93f)

Link to the Reddit discussion: [[URL to the discussion](https://www.reddit.com/r/Anki/comments/1fh6wgq/theoretical_question_about_fsrs_and_optimisations/)]

**Comment by L-M-Sherlock** [comment]
It's challenging to maximize the retention on a specific date. The universal method is reinforcement learning, but it's impractical in Anki.

**Comment by cjdduarte** [comment]
I completely agree, and I trust the algorithm's effectiveness 100%. However, when someone has already been studying for at least a year and knows they’ll be taking an exam in 6 or 9 months, the algorithm could potentially be adjusted to converge toward that specific goal. Naturally, the workload would increase as the exam date approaches, but the user would already be aware of that "risk."

This is just a suggestion, but it’s a challenge worth exploring. I believe it could be a valuable feature for those who use Anki specifically to prepare for exams.

**Comment by cjdduarte** [comment]
Would it be possible to "personalize" the FSRS in a way similar to how we use custom fields in Anki? Maybe allowing users to adjust the algorithm to focus on specific dates like exam deadlines. Sometimes, users themselves might be able to try implementing this with a bit of guidance or through an open framework. How could we approach that? Would it involve creating custom settings within FSRS, or could we somehow modify existing intervals manually in a more efficient way?

**Comment by L-M-Sherlock** [comment]
You need to ask Dae because I'm far from an expert on Anki's framework. I guess it requires to refactor the Anki's code massively.
